src/Client.php

```diff
@@ -339,7 +339,16 @@ private function applyOptions(RequestInterface $request, array &$options)
 
         if (!empty($options['auth']) && is_array($options['auth'])) {
             $value = $options['auth'];
-            $type = isset($value[2]) ? strtolower($value[2]) : 'basic';
+            $type = 'basic';
+            if (isset($value['type'])) {
+                $type = strtolower($value['type']);
+            } else if (isset($value[2])) {
+                $type = strtolower($value[2]);
+            }
+            $value = [
+                array_key_exists('username', $value) ? $value['username'] : $value[0],
```

Comment:
i'd like to maintain behavioral consistency, and the previous implementation does not restrict null. i think null is acceptable.